stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -59,18 +59,33 @@ end
 @inline mul!(y::StridedVector{T}, A::StridedVecOrMat{T}, x::StridedVector{T},
              alpha::Number, beta::Number) where {T<:BlasFloat} =
     gemv!(y, 'N', A, x, alpha, beta)
-# Complex matrix times real vector. Reinterpret the matrix as a real matrix and do real matvec compuation.
-for elty in (Float32,Float64)
+for elty in (Float32, Float64)
     @eval begin
+        # Complex matrix times real vector. Reinterpret the matrix as a real matrix and do real matvec compuation.
         @inline function mul!(y::StridedVector{Complex{$elty}}, A::StridedVecOrMat{Complex{$elty}}, x::StridedVector{$elty},
-                              alpha::Real, beta::Real)
+                alpha::Real, beta::Real)
             Afl = reinterpret($elty, A)
             yfl = reinterpret($elty, y)
             mul!(yfl, Afl, x, alpha, beta)
             return y
         end
+        # Real matrix times complex vector.
+        # Multiply the matrix with the real and imaginary parts separately
+        @inline function mul!(y::StridedVector{Complex{$elty}}, A::StridedMatrix{$elty}, x::StridedVector{Complex{$elty}},
```

Comment:
How about replacing `A::StridedMatrix{$elty}` with `A::StridedMaybeAdjOrTransMat{$elty}`

Reply:
```julia
julia> M = transpose(rand(Float32, 1000, 1000)); V = rand(ComplexF32, size(M,2));

julia> C1 = @btime $M * $V;
  1.593 ms (1 allocation: 7.94 KiB)

julia> Revise.track(LinearAlgebra)

julia> C1 = @btime $M * $V;
  297.964 μs (1 allocation: 7.94 KiB)
```

I'll update this